### DISCUSSION TITLE
Does the warpx any way to sample electric field at some points or a 1d array in some direction with the moving windows open?

### DISCUSSION THREAD

**Discussion by jiecaix** [BLAST-WarpX/warpx#4434] [Q&A] [answered]
I want to sample some points electric field to analyisis it, so i want to know if it possible to direction write down the slice of field data at some fixed points or vector array points directly at simulation.

**Comment by jiecaix** [comment]
Does the CallBack function in python could read the field data and write it down?

**Comment by ax3l** [comment, marked as answer]
Hi,

This should work with the `FieldProbe` diagnostics with any `.probe_geometry` as long as `<reduced_diags_name>.do_moving_window_FP = 1` is also set.
https://warpx.readthedocs.io/en/latest/usage/parameters.html#reduced-diagnostics

This will only move the initial positions of the probes relative to the moving window. Do you need something more complicated?
